management-core/src/main/java/com/ns/task/service/ProductCoreServiceImpl.java

```diff
@@ -38,9 +40,15 @@ public ProductCoreServiceImpl(ModelMapper mapper, ProductRepository repository)
     @Override
     public ProductDTO insertProduct(ProductDTO product) {
         ProductEntity productEntity = convertToEntity(product);
-        logger.debug("Sending data to DB {}", productEntity);
-        ProductEntity productSaved = repository.saveProduct(productEntity);
-        logger.debug("Getting persisted data from insert to DB {}", productSaved);
+        ProductEntity productSaved = new ProductEntity();
+        try {
+            logger.debug("Sending data to DB {}", productEntity);
+            productSaved = repository.saveProduct(productEntity);
+            logger.debug("Getting persisted data from insert to DB {}", productSaved);
+        } catch (SQLIntegrityConstraintViolationException integrity) {
+            logger.debug("An exception has been thrown");
+        }
```

Comment:
If there's an exception in the insertion, you're still retrieving the DTO.

Is this the expected flow?

Reply:
Here I'm not catching anything anymore, just throwing the exception (which is unchecked by the way) because I have some doubts according to which approach will take in order to handle exceptions, I can let as the last one in my task in where I return the same dto but within there is an object call metadata in where I return information related to the exception,but I do not know if this is a best practice or there are a better approach, what you think about?

Reply:
Fixed, now I am throwing a custom exception